src/containers/MasterWindow.js

```diff
@@ -63,146 +63,125 @@ class MasterWindowContainer extends Component {
   async onWebsocketEvent(event) {
     const { includedTabsInfo, stale } = event;
 
-    if (stale) {
-      // some tabs data got updated/row was added
-      if (includedTabsInfo) {
-        const rowsRequests = this.getTabRowsRequests(includedTabsInfo);
+    const activeTab = includedTabsInfo
+      ? Object.values(includedTabsInfo).find((tabInfo) =>
+          this.isActiveTab(tabInfo.tabId)
+        )
+      : null;
 
-        // wait for all the rows requests to finish
-        return await Promise.all(rowsRequests).then((res) => {
-          this.mergeDataIntoIncludedTabs(res);
-        });
-      } else {
-        this.fireFullUpdateData();
-      }
-    }
+    // Document header got staled
+    if (stale) {
+      const { params, fireUpdateData } = this.props;
 
-    if (includedTabsInfo) {
-      const tabIds = Object.keys(includedTabsInfo);
-      this.refreshActiveTab(tabIds);
+      fireUpdateData({
+        windowId: params.windowType,
+        documentId: params.docId,
+        doNotFetchIncludedTabs: true,
+      });
     }
-  }
-
-  fireFullUpdateData() {
-    const { params, fireUpdateData } = this.props;
-
-    fireUpdateData(
-      'window',
-      params.windowType,
-      params.docId,
-      null,
-      null,
-      null,
-      null
-    );
-  }
 
-  getTabRowsRequests(includedTabsInfo) {
-    const requests = [];
-
-    forEach(includedTabsInfo, (tab, tabId) => {
-      if (this.isActiveTab(tabId)) {
-        const { staleRowIds } = tab;
+    // Active included tab got staled
+    if (activeTab) {
+      // Full tab got staled
+      if (activeTab.stale) {
+        this.refreshActiveTab();
+      }
+      // Some included rows got staled
+      else {
+        const { staleRowIds } = activeTab;
         staleRowIds.forEach((rowId) => {
-          requests.push(this.getTabRow(tabId, rowId));
```

Comment:
I'm almost sure this is going to cause trouble when more than one row is changed at a time since all of my async code for grabbing rows was removed. Not sure if it was intentional but I'm pretty sure we'll need that later.

Reply:
@siemiatj  fuck, agree... is that because I took out the Promise.all(...).then(...) thing?

btw, some while ago I've introduced an endpoint which allows you to fetch all rows by IDs together

> .../window/{windowId}/{documentId}/{tabId}?ids=1,2,3

i.e. that endpoint which allows you to fetch the whole tab now it is accepting the `ids=<comma separated ids>` param.

Reply:
https://github.com/metasfresh/metasfresh-webui-frontend/pull/2671